fish-rust/src/builtins/shared.rs

```diff
@@ -23,17 +23,50 @@ mod builtins_ffi {
             streams: Pin<&mut io_streams_t>,
             cpp_args: &Vec<wcharz_t>,
             builtin: RustBuiltin,
-        );
+            status_code: &mut i32,
+        ) -> bool;
```

Comment:
CXX doesn't support `std::optional` but we can return a makeshift `SharedPtr<i32>`, which is probably nicer than an output parameter

Reply:
What would that look like on the Rust side? How do I return a `SharedPtr`?

Reply:
[SharedPtr docs here](https://cxx.rs/binding/sharedptr.html), but I think this is fine as written.